### DISCUSSION TITLE
How to make production requests with getStaticProps to an /api/any-lambda?

### DISCUSSION THREAD

**Discussion by omariosouto** [vercel/next.js#16721] [Help]
Is it possible to make server side requests to a NextJS API Route?

**Comment by jamesmosier** [comment]
It is not possible because during builds inside of `getStaticProps` (or `getStaticPaths`) because the API route doesn't exist yet because the code is still being built. I suppose if you already have `/api/any-lambda` hosted somewhere you could call that endpoint but it would be calling a previously deployed version of that endpoint and not the latest one.

Instead, [Next recommends](https://nextjs.org/docs/basic-features/data-fetching#write-server-side-code-directly) writing the server code directly in your `getStaticProps` function.
